beacon-chain/blockchain/execution_engine.go

```diff
@@ -232,7 +232,10 @@ func (s *Service) notifyNewPayload(ctx context.Context, postStateVersion int,
 			"blockRoot":    fmt.Sprintf("%#x", root),
 			"invalidCount": len(invalidRoots),
 		}).Warn("Pruned invalid blocks")
-		return false, ErrInvalidPayload
+		return false, invalidBlock{
+			ancestorRoots: invalidRoots,
+			error:         ErrInvalidPayload,
```

Comment:
Is there a problem by not returning the block root here? I see anyway in `ReceiveBlock` we will have it and mark it as bad.

Reply:
No problem. It's the same as before. When it's zeros then we use the self-root